community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -154,32 +158,47 @@ public Property nodeSetProperty( KernelStatement state, long nodeId, DefinedProp
         return entityWriteOperations.nodeSetProperty( state, nodeId, property );
     }
 
-    private void validateNoExistingNodeWithLabelAndProperty(
+    private ExactPredicate[] getAllPropertyValues( KernelStatement state, SchemaDescriptor schema, NodeItem node )
+    {
+        int[] propertyIds = schema.getPropertyIds();
+        ExactPredicate[] values = new ExactPredicate[propertyIds.length];
+        for ( int i = 0; i < values.length; i++ )
+        {
+            Object value = nodeGetProperty( state, node, propertyIds[i] );
```

Comment:
I think it'll be faster to implement this in terms of the `Cursor<PropertyItem>` from `nodeGetProperties`.